common/etc/nginx/include/s3gateway.js

```diff
@@ -119,6 +132,15 @@ function awsHeaderDate(r) {
     return _amzDatetime(now, _eightDigitDate(now));
 }
 
+function readCredentials() {
+    try {
+        var creds = fs.readFileSync('/tmp/credentials.json');
```

Comment:
Unit tests fail when the file is `/tmp/credentials.json` does not exist (which would be the default state in a unit test run). We need to figure out a way to mock the file read or to add a file with fake values within the execution of the tests.

Reply:
Hmm, this file being missing shouldn't be any problem, really. I saw the failing test here:
https://github.com/nginxinc/nginx-s3-gateway/runs/5117512199?check_suite_focus=true
However, looking at it doesn't give me any clue why they are failing. Do you have an exact line number in the logs where the reason for the failing test is given?

I don't think that mocking the file is a good idea from the test perspective. The test shouldn't care what temporary storage the implementation is using to cache credentials. It could easily change to environment variables at some point. The test should then not need to change just because we might change the implementation. I think it is better to find the issue why the code can not work without this file being present (in my tests in ECS, it worked pretty well and the file was not there on the first request, either).